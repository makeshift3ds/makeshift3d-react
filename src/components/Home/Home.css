@import '../../assets/css/globals.css';

article {
  flex: 1 0 auto;
  margin-right: 1rem;
  padding: 1rem;
}

aside {
  width: 300px;
  padding: 1rem;
  
  & .aside-block {
    background-color: var(--tan);
    border-radius: 4px;
    box-shadow: 1px 1px 2px #000;
    color: gray(50);
    padding: 1rem;

    & span {
      color: gray(100);
    }
  }
}
