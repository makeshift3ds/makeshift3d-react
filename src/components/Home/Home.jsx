import React from 'react';
import { Markup } from 'interweave';
import 'babel-polyfill';

import './Home.css';

export default class Home extends React.Component {
  async componentDidMount() {
    try {
      const res = await fetch('http://localhost:3000/pages/foobar');
      const page = await res.json();
      this.setState({
        page,
      });
    } catch (e) {
      console.log('error', e);
    }
  }

  state = {
    page: {},
  };

  render() {
    const { title, body } = this.state.page;

    return (
      <div className="content-container">
        <article>
          <h1>{title}</h1>
          <Markup content={body} />
        </article>
        <aside>
          <div className="aside-block">
            <p>This guy codes!</p>
            <span> - Random Redditor</span>
          </div>
        </aside>
      </div>
    );
  }
}

