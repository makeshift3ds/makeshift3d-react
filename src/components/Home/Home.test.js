import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home.jsx';

describe('<Home />', () => {
  it('should render 1 <Home /> component', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render an article element', () => {
    const component = shallow(<Home />);
    expect(component.find('article')).toHaveLength(1);
  });

  it('should render an aside element', () => {
    const component = shallow(<Home />);
    expect(component.find('aside')).toHaveLength(1);
  });

  it('should match snapshot', () => {
    const component = shallow(<Home />);
    expect(component).toMatchSnapshot();
  });
});
