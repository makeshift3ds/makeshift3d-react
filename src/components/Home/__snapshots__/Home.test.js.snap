// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article>
          <h1 />
          <Markup
            content=""
            disableLineBreaks={false}
            disableWhitelist={false}
            emptyContent={null}
            noHtml={false}
            noHtmlExceptMatchers={false}
            tag="div"
            tagName="span"
          />
        </article>,
        <aside>
          <div
            className="aside-block"
          >
            <p>
              This guy codes!
            </p>
            <span>
               - Random Redditor
            </span>
          </div>
          <div
            className="aside-block"
          >
            <p>
              I use PERL to write code that no one understands.
            </p>
            <span>
               - Ken Elliott
            </span>
          </div>
        </aside>,
      ],
      "className": "content-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1 />,
            <Markup
              content=""
              disableLineBreaks={false}
              disableWhitelist={false}
              emptyContent={null}
              noHtml={false}
              noHtmlExceptMatchers={false}
              tag="div"
              tagName="span"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "content": "",
              "disableLineBreaks": false,
              "disableWhitelist": false,
              "emptyContent": null,
              "noHtml": false,
              "noHtmlExceptMatchers": false,
              "tag": "div",
              "tagName": "span",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="aside-block"
            >
              <p>
                This guy codes!
              </p>
              <span>
                 - Random Redditor
              </span>
            </div>,
            <div
              className="aside-block"
            >
              <p>
                I use PERL to write code that no one understands.
              </p>
              <span>
                 - Ken Elliott
              </span>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  This guy codes!
                </p>,
                <span>
                   - Random Redditor
                </span>,
              ],
              "className": "aside-block",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "This guy codes!",
                },
                "ref": null,
                "rendered": "This guy codes!",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " - Random Redditor",
                },
                "ref": null,
                "rendered": " - Random Redditor",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  I use PERL to write code that no one understands.
                </p>,
                <span>
                   - Ken Elliott
                </span>,
              ],
              "className": "aside-block",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "I use PERL to write code that no one understands.",
                },
                "ref": null,
                "rendered": "I use PERL to write code that no one understands.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " - Ken Elliott",
                },
                "ref": null,
                "rendered": " - Ken Elliott",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "aside",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article>
            <h1 />
            <Markup
              content=""
              disableLineBreaks={false}
              disableWhitelist={false}
              emptyContent={null}
              noHtml={false}
              noHtmlExceptMatchers={false}
              tag="div"
              tagName="span"
            />
          </article>,
          <aside>
            <div
              className="aside-block"
            >
              <p>
                This guy codes!
              </p>
              <span>
                 - Random Redditor
              </span>
            </div>
            <div
              className="aside-block"
            >
              <p>
                I use PERL to write code that no one understands.
              </p>
              <span>
                 - Ken Elliott
              </span>
            </div>
          </aside>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1 />,
              <Markup
                content=""
                disableLineBreaks={false}
                disableWhitelist={false}
                emptyContent={null}
                noHtml={false}
                noHtmlExceptMatchers={false}
                tag="div"
                tagName="span"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "content": "",
                "disableLineBreaks": false,
                "disableWhitelist": false,
                "emptyContent": null,
                "noHtml": false,
                "noHtmlExceptMatchers": false,
                "tag": "div",
                "tagName": "span",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="aside-block"
              >
                <p>
                  This guy codes!
                </p>
                <span>
                   - Random Redditor
                </span>
              </div>,
              <div
                className="aside-block"
              >
                <p>
                  I use PERL to write code that no one understands.
                </p>
                <span>
                   - Ken Elliott
                </span>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    This guy codes!
                  </p>,
                  <span>
                     - Random Redditor
                  </span>,
                ],
                "className": "aside-block",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "This guy codes!",
                  },
                  "ref": null,
                  "rendered": "This guy codes!",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " - Random Redditor",
                  },
                  "ref": null,
                  "rendered": " - Random Redditor",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    I use PERL to write code that no one understands.
                  </p>,
                  <span>
                     - Ken Elliott
                  </span>,
                ],
                "className": "aside-block",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "I use PERL to write code that no one understands.",
                  },
                  "ref": null,
                  "rendered": "I use PERL to write code that no one understands.",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " - Ken Elliott",
                  },
                  "ref": null,
                  "rendered": " - Ken Elliott",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "aside",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
